File: competition/06-string/string.js
const parts = new Array(50000);
for (let i = 0; i < 50000; i++) parts[i] = "item-" + i;
const joined = parts.join(",");
const back = joined.split(",");
const upper = joined.toUpperCase();
console.log(back.length);
console.log(joined.length);
console.log(upper.length);

File: competition/06-string/string.lua
-- String round-trip: build many parts, join, split back, uppercase.
local parts = {}
for i = 0, 49999 do parts[i + 1] = "item-" .. i end
local joined = table.concat(parts, ",")
local back = {}
local k = 0
for tok in string.gmatch(joined, "[^,]+") do
  k = k + 1
  back[k] = tok
end
local upper = string.upper(joined)
print(#back)
print(#joined)
print(#upper)
